File: add/codesnippet/CSharp/e676da23-14f2-44cc-b84b-_1.cs
      try {
         Match match = Regex.Match(input, pattern, options,
                                   TimeSpan.FromSeconds(1));
         while (match.Success) {
               // Handle match here...
   
               match = match.NextMatch();
         }  
      }
      catch (RegexMatchTimeoutException) {
         // Do nothing: assume that exception represents no match.
      }

File: add/codesnippet/VisualBasic/e676da23-14f2-44cc-b84b-_1.vb
      Try
         Dim match As Match = Regex.Match(input, pattern, options, 
                                          TimeSpan.FromSeconds(1))
         Do While match.Success
               ' Handle match here...
   
               match = match.NextMatch()
         Loop  
      Catch e As RegexMatchTimeoutException
         ' Do nothing: assume that exception represents no match.
      End Try   